### ISSUE TITLE
Crash while using an editor

### ISSUE THREAD

**Issue by killio99** [4ian/GDevelop#1196]
=> Please write here a short description of when the error occured and how to reproduce it.
You also may have to create an account on GitHub before posting.

When you're ready, click on "Submit new issue". Don't change the rest of the message. Thanks!

## Error stack (don't write anything here)
```
Error: Failed to execute 'getImageData' on 'CanvasRenderingContext2D': The source width is 0.
    at Function.e.measureFont (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/0.b173112f.chunk.js:1:714033)
    at Function.e.measureText (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/0.b173112f.chunk.js:1:711088)
    at t.updateText (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/0.b173112f.chunk.js:1:1305689)
    at t.get (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/0.b173112f.chunk.js:1:1310055)
    at i.update (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:65835)
    at new i (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:64050)
    at Object.createNewInstanceRenderer (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:523958)
    at e.getRendererOfInstance (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:170950)
    at InitialInstanceJSFunctor.instancesRenderer.invoke (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:171998)
    at Array.<anonymous> (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/libGD.js:20:19250)
    at _emscripten_asm_const_iii (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/libGD.js:20:28719)
    at Array.pM (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/libGD.js:28:36778)
    at Array.oM (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/libGD.js:28:36736)
    at Wba (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/libGD.js:26:159333)
    at sz (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/libGD.js:24:167398)
    at InitialInstancesContainer.IterateOverInstancesWithZOrdering.InitialInstancesContainer.IterateOverInstancesWithZOrdering [as iterateOverInstancesWithZOrdering] (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/libGD.js:8718:3)
    at e.value (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:172849)
    at e.value (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:362518)
    at t.e._renderScene (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:382603)
    at t.value (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:386568)
    at t.value (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/1.568af440.chunk.js:1:383269)
    at Fa (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:218823)
    at Va (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:221499)
    at file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:234728
    at Object.t.unstable_runWithPriority (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:245473)
    at As (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:234663)
    at Rs (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:234439)
    at Ns (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:233784)
    at Ds (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:235159)
    at Tn (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:166685)
    at HTMLDocument.i (file:///C:/Users/Owner/AppData/Local/Programs/gdevelop/resources/app.asar/www/static/js/55.b8eb2518.chunk.js:1:669971)
```

## Component stack (don't write anything here)
```

    in t
    in div
    in n
    in div
    in t
    in l
    in div
    in div
    in t
    in div
    in t
    in Unknown
    in t
    in t
    in div
    in t
    in t
    in n
    in t
    in Unknown
    in div
    in div
    in t
    in div
    in div
    in t
    in t
    in t
    in div
    in t
    in t
    in n
    in t
    in t
    in t
    in t
    in t
    in t
    in t
    in t
    in n
    in t
    in t
    in t
    in DragDropContext(t)
    in t
    in t
```

**Comment by 4ian** [comment]
Are you using beta79?

**Comment by Bouh** [comment]
When font size is set to 0 we have this error, not crash for me but exactlly same error in console from preview.

It would be better to have more details on the bug reports generated.
I found this bug at random I just knew it was affecting the texts because the report says "at Function.e.measureFont"

**Comment by Bouh** [comment]
I've looked the previous crashs and lot of crashs seem have same report log about text object.
Font missing, or Font size 0, 
Because this two parameters haven't defaut value ?

**Comment by killio99** [comment]
Not sure. How do I check?

**Comment by 4ian** [comment]
I've done fixes in e5aac3d75dbdad916f459cb078111cf297be01ec to have the character size being 1 at the minimum - both enforced by the editor and by the rendered instances/rendered object during the game so that both the editor and the game are robust against wrong values (values <= 0).

**Comment by 4ian** [comment]
Closing this for now, if this happen again later in the next versions we can re-open or have another issue.
